docs: add upgrade issue from v1.4 to v1.5 (#780)
harvester/docs: docs/upgrade/v1-4-2-to-v1-5-0.md

Related issues:
- [[BUG] AirGap Upgrades Seem Blocked with Fluentbit/FluentD](https://github.com/harvester/harvester/issues/7955)

> ### 3. Upgrade stuck on waiting for Harvester bundle

Author: BTW, I copied and modified it from here https://docs.harvesterhci.io/v1.5/upgrade/v1-3-2-to-v1-4-0#2-upgrade-stuck-on-waiting-for-harvester-bundle.

> ### 3. Upgrade stuck on waiting for Harvester bundle

Author: It seems we had provided this way to users before?

> ### 3. Upgrade stuck on waiting for Harvester bundle

Reviewer: A troubleshooting is guide the user to follow `1,2,3,4` and then confirm &/ fix some thing. Just assume the user has very basic kubectl skill.

Back to this, we can organize the info like:

1.If your cluster is stucking on `upgrade system service`
     Optional: with an image from the stucking cluster

2.Following info is observed on the `harvester-system/apply-manifest-*` pod log,

```
...
managedchart.management.cattle.io/harvester-crd patched
Waiting for ManagedChart fleet-local/harvester from generation 51
Target version: 1.5.0, Target state: ready
Current version: 1.5.0, Current state: null, Current generation: 51  // loop and does not finish
Sleep for 5 seconds to retry
...
```

3. You also observe below info, then your cluster hit this issue

```
> kubectl get ManagedChart -n fleet-local harvester -o yaml
...
      modifiedStatus:
      - apiVersion: kubevirt.io/v1
        kind: KubeVirt
        name: kubevirt
        namespace: harvester-system
        patch: '{"spec":{"configuration":{"vmStateStorageClass":"vmstate-persistence"}}}'
      name: fleet-local/local
```

The workaround fix the issue:
...

> This occurs when you upgrade harvester from old vesrions, including `v1.0.x`, `v1.1.x` and `v1.2.x`.

Author: It must happen. If your initial Harvester version is v1.0.x, v1.1.x or v1.2.x, then you try to upgrade it one version by one version. Then, you'll get the error when upgrading to v1.5.0.

> When you upgrade from an old Harvester version (such `v1.0.x`, `v1.1.x`, and `v1.2.x`), the upgrade process may become stuck on waiting for the `mcc-harvester` bundle CR to become ready.

Reviewer 2: ```suggestion
When you upgrade from an old Harvester version (such as `v1.0.x`, `v1.1.x`, and `v1.2.x`), the upgrade process may become stuck on waiting for the `mcc-harvester` bundle CR to become ready.
```

```shell
> kubectl get bundles -n fleet-local
NAME                                          BUNDLEDEPLOYMENTS-READY   STATUS
mcc-harvester                                 0/1                       Modified(1) [Cluster fleet-local/local]; kubevirt.kubevirt.io harvester-system/kubevirt modified {"spec":{"configuration":{"vmStateStorageClass":"vmstate-persistence"}}}
```

> The root cause is that we didn't apply the latest `depdendency_charts` CRDs because Helm doesn't manage CRDs for us. In order to make upgrade continue, please execute folloing script. It includs `kubevirt`, `csi-snapshotter` and `whereabouts`.

Reviewer 3: ```suggestion
The root cause is that we didn't apply the latest `depdendency_charts` CRDs because Helm doesn't manage CRDs for us. In order to make upgrade continue, please execute following script. It includes `kubevirt`, `csi-snapshotter` and `whereabouts`.
```

```shell

> kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/refs/heads/v1.5/deploy/charts/harvester/dependency_charts/kubevirt-operator/crds/crd-kubevirt.yaml

Reviewer 4: we should use the tagged version here

kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/refs/tags/v1.5.0/deploy/charts/harvester/dependency_charts/whereabouts/crds/whereabouts.cni.cncf.io_ippools.yaml
kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/refs/tags/v1.5.0/deploy/charts/harvester/dependency_charts/whereabouts/crds/whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml
```

After 5 minutes, please check status in `mcc-harvseter` of `bundle.fleet.cattle.io/v1alpha1`. If it still shows the same error, we need to force bundle to resync it.

```shell
#!/bin/bash

patch_fleet_bundle() {
  local bundleName=$1
  local generation=$(kubectl get -n fleet-local bundle ${bundleName} -o jsonpath='{.spec.forceSyncGeneration}')
  local new_generation=$((generation+1))
  patch_manifest="$(mktemp)"
  cat > "$patch_manifest" <<EOF
{
  "spec": {
    "forceSyncGeneration": $new_generation
  }
}
EOF
  echo "patch bundle to new generation: $new_generation"
  kubectl patch -n fleet-local bundle ${bundleName}  --type=merge --patch-file $patch_manifest
  rm -f $patch_manifest
}

for bundle in mcc-harvester
do
  patch_fleet_bundle ${bundle}
done
```

> Besides, there is another case that gets stuck. Please make sure that the cdi CRD exists. 

Reviewer 3: ```suggestion
Besides, there is another case that gets stuck. Please make sure that the `cdi` CRD exists. 
```
